onadata/apps/data_migration/surveytree.py

```diff
@@ -109,3 +60,59 @@ def insert_field_into_group_chain(self, field, group_chain):
             parent = group_field
 
         parent.append(field)
+
+    def sort(self, xformtree):
```

Comment:
First thought was that I need to preserve the order because otherwise `formpack` parser fails - then it turned out that it's about field duplicates, and it's quite easy to get rid of duplicated fields if you have them sorted first. I've attached now sort method to `data_migrator` steps.